crates/sui-indexer/src/models/packages.rs

```diff
@@ -6,37 +6,31 @@ use crate::schema::packages;
 
 use diesel::prelude::*;
 
-use sui_json_rpc_types::SuiMovePackage;
-use sui_types::base_types::{ObjectID, SuiAddress};
+use crate::models::objects::NamedBcsBytes;
+use sui_json_rpc_types::SuiRawMovePackage;
+use sui_types::base_types::SuiAddress;
 
 #[derive(Queryable, Insertable, Debug, Identifiable)]
-#[diesel(table_name = packages)]
+#[diesel(table_name = packages, primary_key(package_id, version))]
 pub struct Package {
-    pub id: Option<i64>,
     pub package_id: String,
+    pub version: i64,
     pub author: String,
-    pub module_names: Vec<String>,
-    pub package_content: String,
+    pub data: Vec<NamedBcsBytes>,
 }
 
 impl Package {
-    pub fn try_from(
-        id: ObjectID,
-        sender: SuiAddress,
-        package: &SuiMovePackage,
-    ) -> Result<Self, IndexerError> {
+    pub fn try_from(sender: SuiAddress, package: &SuiRawMovePackage) -> Result<Self, IndexerError> {
         Ok(Self {
-            id: None,
-            package_id: id.to_hex(),
+            package_id: package.id.to_string(),
+            version: package.version.value() as i64,
             author: sender.to_string(),
-            module_names: package.disassembled.keys().cloned().collect(),
-            // TODO: store raw package bytes instead when object refactoring is done.
-            package_content: serde_json::to_string(&package.disassembled).map_err(|err| {
-                IndexerError::InsertableParsingError(format!(
-                    "Failed converting package module map to JSON with error: {:?}",
-                    err
-                ))
-            })?,
+            data: package
+                .module_map
+                .clone()
+                .into_iter()
+                .map(|(k, v)| NamedBcsBytes(k, v))
```

Comment:
not for this PR, I am thinking of storing one module's BCS in one column instead of BCS arrays for the whole package, as
- as we will have getMoveModule and getMoveFunction, always bcs_from the whole package is not efficient, the related map can be memory-heavy as well
- iiuc we can always construct a `package` based on package ID and its modules, but cannot construct module easily with functions b/c of structs etc.

I was also thinking of field size, where limit is 1GB, https://www.postgresql.org/docs/current/limits.html, but that should be fine as long as we have reasoable object size limit, wdyt?

Reply:
The core team is working on object size limit? I don't think move package will ever exceed 1GB, so we should be ok?

wrt storing module separately, we can use unnest instead
```
SELECT *
FROM (SELECT object_id, UNNEST(bcs) AS module
      FROM objects
      WHERE object_id = '0x0000000000000000000000000000000000000000000000000000000000000002') as o
WHERE (o.module).name = 'coin'
```
this will give you just the bcs bytes for that module
<img width="1178" alt="image" src="https://user-images.githubusercontent.com/1888654/223765698-1ad2b81d-cafd-4631-8139-eb3563715b0f.png">